File: Programmers/Level 2/구명보트.js
function solution(people, limit) {
    var answer = 0
    people = people.sort((a, b) => b - a)
    let [left, right] = [0, people.length-1]
    while (left < right) {
        if (people[left] + people[right] > limit) {
            left += 1
        } else {
            left += 1
            right -= 1
        }
        answer += 1
    }
    if (left === right) answer += 1
    return answer
}

File: Programmers/Level 2/구명보트.py
def solution(people, limit):
	answer = 0
	people = sorted(people, reverse=True)
	left, right = 0, len(people)-1
	while left < right:
		if people[left] + people[right] <= limit:
			right -= 1
		left += 1
		answer += 1
	return answer if left != right else answer + 1


# print(solution([70, 50, 80, 50], 100))  # 3
# print(solution([70, 80, 50], 100))  # 3
